pkgs/development/python-modules/scipy/default.nix

```diff
@@ -67,6 +95,15 @@ buildPythonPackage rec {
   #
   hardeningDisable = lib.optionals (stdenv.isAarch64 && stdenv.isDarwin) [ "stackprotector" ];
 
+  dontUsePipBuild = true;
+  pypaBuildFlags = [
+    # Skip sdist
+    "--wheel"
+
+    "-Csetup-args=-Dblas=cblas"
+    "-Csetup-args=-Dlapack=lapacke"
+  ];
```

Comment:
Apparently, `pip` now supports this as well: https://github.com/mesonbuild/meson-python/issues/415#issuecomment-1536532720

However, we should probably finish the `build` version first, and then move back to `pip`, because if we're to introduce `pipWheelFlags` we'll have to go through staging

Reply:
@FRidh OK, new plan then:

- drop `-m build` (because there's no need for it),
- introduce `pipWheelFlags`,
- expose `setupHook` in `python3Packages.meson-python`, so that including `meson-python` in `nativeBuildInputs` is sufficient to build meson-python-backed packages
- `meson-python.setupHook` looks at the same variables as `meson`, e.g. `mesonFlags`
- `meson-python.setupHook` prepends `mesonFlags` with `-Csetup-args=` and extends `pipBuildFlags` with them

Does that sounds OK? I fear this is too much indirection and could be fragile

Reply:
Yes, I think this is the right approach.

Note we'd want this also in the future when we would replace `pip` with `build`.

Reply:
> @FRidh OK, new plan then:
> 
>   * drop `-m build` (because there's no need for it),
>   * introduce `pipWheelFlags`,

I'm working on `pipBuildFlags` at https://github.com/NixOS/nixpkgs/pull/239969 .

>   * expose `setupHook` in `python3Packages.meson-python`, so that including `meson-python` in `nativeBuildInputs` is sufficient to build meson-python-backed packages
>   * `meson-python.setupHook` looks at the same variables as `meson`, e.g. `mesonFlags`
>   * `meson-python.setupHook` prepends `mesonFlags` with `-Csetup-args=` and extends `pipBuildFlags` with them
> 
> Does that sound OK? I fear this is too much indirection and could be fragile

That sounds great to me too, but what I learned from the experience in https://github.com/NixOS/nixpkgs/pull/239969 is that passing `--cross-file` to meson is what's needed.

Reply:
So, does `--cross-file` go into `mesonFlags` (`pipBuildFlags`) in addition to `-Dblas...`?